sysgen/sysgen.go

```diff
@@ -532,6 +532,44 @@ func generateArg(
 			skipSyscall(fmt.Sprintf("missing const %v", a[0]))
 		}
 		fmt.Fprintf(out, "&ConstType{%v, TypeSize: %v, BigEndian: %v, Val: uintptr(%v)}", common(), size, bigEndian, val)
+	case "proc":
+		canBeArg = true
+		size := uint64(ptrSize)
+		bigEndian := false
+		var valuesStart string
+		var valuesPerProc string
+		if isField {
+			if want := 3; len(a) != want {
+				failf("wrong number of arguments for %v arg %v, want %v, got %v", typ, name, want, len(a))
+			}
+			size, bigEndian = decodeIntType(a[0])
+			valuesStart = a[1]
+			valuesPerProc = a[2]
+		} else {
+			if want := 2; len(a) != want {
+				failf("wrong number of arguments for %v arg %v, want %v, got %v", typ, name, want, len(a))
+			}
+			valuesStart = a[0]
+			valuesPerProc = a[1]
+		}
+		valuesStartInt, err := strconv.ParseInt(valuesStart, 10, 64)
+		if err != nil {
+			failf("couldn't parse '%v' as int64", valuesStart)
+		}
+		valuesPerProcInt, err := strconv.ParseInt(valuesPerProc, 10, 64)
+		if err != nil {
+			failf("couldn't parse '%v' as int64", valuesPerProc)
+		}
+		if valuesPerProcInt < 1 {
+			failf("values per proc '%v' should be >= 1", valuesPerProcInt)
+		}
+		if valuesStartInt >= (1 << (size * 8)) {
+			failf("values starting from '%v' overflow desired type of size '%v'", valuesStartInt, size)
+		}
+		if valuesStartInt + 16 * valuesPerProcInt >= (1 << (size * 8)) {
```

Comment:
Alex uses 20 procs. Let's make the max 32. And check that procs parameter <= 32 in config package.